junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClassNameFilterUtil.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apiguardian.api.API;
+
+/**
+ * Class-related predicate holder used by execution listener, execution condition predicates
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ */
+@API(status = INTERNAL, since = "5.5")
+public class ClassNameFilterUtil {
```

Comment:
⭕️ I think we should write a unit test for this class now. It should probably have existed earlier. 😉

Reply:
Added UnitTests for the class